--- lib/MP3/Tag/ID3v1.pm.orig	2009-11-27 23:55:36.000000000 -0500
+++ lib/MP3/Tag/ID3v1.pm	2014-08-24 22:36:04.898266050 -0400
@@ -499,11 +499,11 @@
 			'Techno-Industrial', 'Electronic', 'Pop-Folk', 'Eurodance', 'Dream',
 			'Southern Rock', 'Comedy', 'Cult', 'Gangsta', 'Top 40', 
 			'Christian Rap', 'Pop/Funk', 'Jungle', 'Native American', 'Cabaret', 'New Wave',
 			'Psychadelic', 'Rave', 'Showtunes', 'Trailer', 'Lo-Fi', 'Tribal',
 			'Acid Punk', 'Acid Jazz', 'Polka', 'Retro', 'Musical', 'Rock & Roll',
-			'Hard Rock', );
+			'Hard Rock');
 
   	@winamp_genres = ( @mp3_genres, 'Folk', 'Folk-Rock', 
 			   'National Folk', 'Swing', 'Fast Fusion', 'Bebob', 'Latin', 'Revival',
 			   'Celtic', 'Bluegrass', 'Avantgarde', 'Gothic Rock',
 			   'Progressive Rock', 'Psychedelic Rock', 'Symphonic Rock',
@@ -519,12 +519,26 @@
 			   'Terror', 'Indie', 'BritPop', 'Negerpunk',
 			   'Polsk Punk', 'Beat', 'Christian Gangsta Rap',
 			   'Heavy Metal', 'Black Metal', 'Crossover',
 			   'Contemporary Christian Music', 'Christian Rock',
 			   'Merengue', 'Salsa', 'Thrash Metal', 'Anime',
-			   'JPop', 'SynthPop',			# 149
-			 ); 
+			   'JPop', 'SynthPop',
+			   # WinAmp 5.6 Genres
+                           'Abstract', 'Art Rock', 'Baroque', 'Bhangra',
+                           'Big Beat', 'Breakbeat', 'Chillout','Downtempo',
+                           'Dub', 'EBM', 'Eclectic', 'Electro', 'Electroclash',
+                           'Emo', 'Experimental', 'Garage', 'Global', 'IDM',
+                           'Illbient', 'Industro-Goth', 'Jam Band', 'Krautrock',
+                           'Leftfield', 'Lounge', 'Math Rock', 'New Romantic',
+                           'Nu-Breakz', 'Post-Punk', 'Post-Rock', 'Psytrance',
+                           'Shoegaze', 'Space Rock', 'Trop Rock', 'World Music',
+                           'Neoclassical', 'Audiobook', 'Audio Theatre',
+                           'Neue Deutsche Welle', 'Podcast', 'Indie Rock',
+                           'G-Funk', 'Dubstep', 'Garage Rock', 'Psybient',
+                           'Reggaestep'
+			 ); 			# 194
+
 }
 
 =pod
 
 =head1 SEE ALSO
