--- setup.py    2017-09-07 11:22:50.887054652 -0400
+++ setup.py    2017-09-07 11:26:05.641134553 -0400
@@ -20,15 +20,10 @@
         try:
             shutil.copy("./rivalcfg/data/99-steelseries-rival.rules", "/var/tmp/portage/sys-apps/rivalcfg-2.6.0/work/rivalcfg-2.6.0/image/lib/udev/rules.d/")
         except IOError:
             print("WARNING: udev rules have not been installed (permission denied)")
             return
-        try:
-            subprocess.call(["udevadm", "trigger"])
-        except OSError:
-            print("WARNING: unable to update udev rules, please run the 'udevadm trigger' command")
-            return
         print("Done!")


 long_description = ""
 if os.path.isfile("README.rst"):
