--- procmeter3-3.6/modules/df.c	2009-04-21 13:26:58.000000000 -0400
+++ procmeter3-3.6/modules/df.c	2010-09-27 09:35:24.591394992 -0400
@@ -15,11 +15,11 @@
 
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-
+#include <errno.h>
 #include <sys/vfs.h>
 
 #include "procmeter.h"
 
 /* The interface information.  */
@@ -289,14 +289,21 @@
          {
           output->graph_value=0;
           strcpy(output->text_value,"not found");
          }
        else if(statfs(disk[i/2],&buf))
-         {
-          output->graph_value=0;
-          strcpy(output->text_value,"statfs error");
-         }
+       {
+	   output->graph_value=0;
+	   if (errno == EOVERFLOW)
+	   {
+	       strcpy(output->text_value,"statfs overflow");
+	   }
+	   else
+	   {
+	       strcpy(output->text_value,"statfs error");
+	   }	   
+       }
        else
          {
           if(i%2)
             {
              long long avail=(buf.f_bavail>>5)*(buf.f_bsize>>5);
