--- modules/longrun.c      2010-12-24 05:40:47.000000000 -0500
+++ modules/longrun.c   2019-12-14 15:34:55.292299114 -0500
@@ -36,11 +36,11 @@
 #define CPUID_TMx86_PROCESSOR_INFO 0x80860001
 #define CPUID_TMx86_FEATURE_LONGRUN(x) ((x) & 0x02)

 static int cpuid_fd = 0;

-static void read_cpuid(loff_t address, int *eax, int *ebx, int *ecx, int *edx) {
+static void read_cpuid(off_t address, int *eax, int *ebx, int *ecx, int *edx) {
         uint32_t data[4];

         if (pread(cpuid_fd, &data, 16, address) != 16) {
                 perror("error reading");
         }
