--- pidgin-2.10.13/libpurple/plugins/ssl/ssl-gnutls.c	2018-03-06 01:00:49.000000000 -0500
+++ pidgin-2.10.13/libpurple/plugins/ssl/ssl-gnutls.c	2020-01-18 12:23:03.999186841 -0500
@@ -401,10 +401,15 @@
 	gnutls_credentials_set(gnutls_data->session, GNUTLS_CRD_CERTIFICATE,
 		xcred);
 
 	gnutls_transport_set_ptr(gnutls_data->session, GINT_TO_POINTER(gsc->fd));
 
+	/* SNI support. */
+	if (gsc->host && !g_hostname_is_ip_address(gsc->host))
+		gnutls_server_name_set(gnutls_data->session, GNUTLS_NAME_DNS, gsc->host, strlen(gsc->host));
+
+
 	gnutls_data->handshake_handler = purple_input_add(gsc->fd,
 		PURPLE_INPUT_READ, ssl_gnutls_handshake_cb, gsc);
 
 	/* Orborde asks: Why are we configuring a callback, then
 	   (almost) immediately calling it?
